aiohttp_security/api.py

```diff
@@ -1,12 +1,53 @@
 import asyncio
+import functools
+
 from aiohttp import web
+
 from aiohttp_security.abc import (AbstractIdentityPolicy,
                                   AbstractAuthorizationPolicy)
 
 IDENTITY_KEY = 'aiohttp_security_identity_policy'
 AUTZ_KEY = 'aiohttp_security_autz_policy'
 
 
+def authorize(required=True, redirect_url=None, permission=None):
+    def wrapper(f):
+        @asyncio.coroutine
+        @functools.wraps(f)
+        def wrapped(*args, **kwargs):
+            # assuming first argument is request
+            assert isinstance(args[0], web.Request)
+            request = args[0]
+
+            # check if coroutine
+            if asyncio.iscoroutinefunction(f):
+                coro = f
+            else:
+                coro = asyncio.coroutine(f)
+
+            # get identity
+            identity = yield from get_user_identity(request)
+            kwargs['identity'] = identity
```

Comment:
Tornado Web Framework for instance passes identity to its template engine by default. I assumed that you'd need the identity for most pages that requires user identity.
I'd appreciate if you could explain why this is bad idea.

Also If you could elaborate why changing function signature by decorator is bad idea, it would also be nice.
I thought changing function signature is what decorators (middlewares) are for and a lot of authentication modules use this design pattern such as in Java, Scala and Go.